apps/tutorial_servo_control_can/TutorialServo/TutorialServo.cpp

```diff
@@ -0,0 +1,31 @@
+#include "mbed.h"
+#include "TutorialServo.h"
+
+TutorialServo::TutorialServo(PinName servoPin, float servoRangeInDegrees = 180.0, 
+                            float minPulsewidthInMs = 1, float maxPulsewidthInMs = 2):
+            m_servoPwmOut(servoPin), 
+            m_servoRangeInDegrees(servoRangeInDegrees),
+            m_minPulsewidthInMs(minPulsewidthInMs),
+            m_maxPulsewidthInMs(maxPulsewidthInMs){
+                m_servoPwmOut = 0;
+            }
+
+TutorialServo::setAngleRangeInDegrees(float degrees){
+    this->servoRangeInDegrees = degrees
+}
+
+TutorialServo::setPulsewidthRangeInMs(float minPulsewidthMs, float maxPulsewidthMs){
+    this->m_minPulsewidthInMs = minPulsewidthMs;
+    this->m_maxPulsewidthInMs = maxPulsewidthMs;
+}
+
+TutorialServo::setPositionInDegrees(float degrees){
+    m_servoPwmOut.pulsewidth(
```

Comment:
Servos can rotate in two directions, the minimum pulse width corresponds to the maximum negative angle, while the maximum pulse width corresponds to the maximum positive angle. So the input to this function, degrees, can be negative. Think about how this affects your conversion algorithm